OsuPlayer.Wpf/ViewModels/CollectionPageViewModel.cs

```diff
@@ -144,10 +144,13 @@ public ICommand DirectPlayCommand
                 {
                     var beatmap = (BeatmapDataModel)param;
                     var map = _beatmapDbOperator.GetBeatmapByIdentifiable(beatmap);
+                    var maps = Beatmaps.Select(m => _beatmapDbOperator.GetBeatmapByIdentifiable(m)).Where(m => m != null).ToList();
+                    var i = maps.FindIndex(m => m == map);
                     if (map == null) return;
                     await PlayController.Default.PlayNewFile(map);
                     await Services.Get<PlayerList>()
-                        .RefreshPlayListAsync(PlayerList.FreshType.All, PlayListMode.RecentList);
+                        .RefreshPlayListAsync(PlayerList.FreshType.All, PlayListMode.Collection, maps);
+                    Services.Get<PlayerList>().Pointer = i == -1 ? 0 : i;
```

Comment:
Here I made some bugs. My original intention is making `Pointer` property's setter private and call `RefreshPlayListAsync()` to refresh the indexes automatically. It's not recommended to set the Pointer explicitly.